terraform/modules/tf_stream_alert_app/iam.tf

```diff
@@ -0,0 +1,117 @@
+// IAM Role: StreamAlert App Execution Role
+resource "aws_iam_role" "stream_alert_app_role" {
+  name               = "${var.function_prefix}_app_role"
+  assume_role_policy = "${data.aws_iam_policy_document.lambda_assume_role_policy.json}"
+}
+
+// IAM Policy Doc: Generic Lambda AssumeRole
+data "aws_iam_policy_document" "lambda_assume_role_policy" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+// IAM Role Policy: Allow the StreamAlert App function to invoke the Rule Processor
+resource "aws_iam_role_policy" "stream_alert_app_role_policy" {
+  name   = "${var.function_prefix}_app_role_policy"
+  role   = "${aws_iam_role.stream_alert_app_role.id}"
+  policy = "${data.aws_iam_policy_document.app_invoke_rule_proc.json}"
+}
+
+// IAM Policy Doc: Allow the StreamAlert App function to invoke the Rule Processor
+data "aws_iam_policy_document" "app_invoke_rule_proc" {
```

Comment:
this policy does more than just invoking the rule processor, I'd split them out into 3 separate policies for clarity and to easily understand the permissions model from console